File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/DepositResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import ut.edu.be_quanlytro.Entity.Enum.DepositStatus;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@Builder
public class DepositResponse {
    private UUID id;
    private UUID roomId;
    private String roomNumber;
    private String phone;
    private String tenantFullName;
    private BigDecimal depositAmount;
    private LocalDate depositDate;
    private LocalDate expectedMoveInDate;
    private String note;
    private DepositStatus status;
}

File: Flutter_QuanLyTro/lib/data/models/response/deposit_response.dart
class DepositResponse {
  final String id;
  final String roomNumber;
  final String tenantFullName;
  final String phone;
  final double depositAmount;
  final String status;
  final String? depositDate;
  final String? expectedMoveInDate;
  final String? note;

  DepositResponse({
    required this.id,
    required this.roomNumber,
    required this.tenantFullName,
    required this.phone,
    required this.depositAmount,
    required this.status,
    this.depositDate,
    this.expectedMoveInDate,
    this.note,
  });

  factory DepositResponse.fromJson(Map<String, dynamic> json) {
    return DepositResponse(
      id: json['id'] ?? '',
      roomNumber: json['roomNumber'] ?? '',
      tenantFullName: json['tenantFullName'] ?? '',
      phone: json['phone'] ?? '',
      depositAmount: (json['depositAmount'] ?? 0).toDouble(),
      status: json['status'] ?? '',
      depositDate: json['depositDate'],
      expectedMoveInDate: json['expectedMoveInDate'],
      note: json['note'],
    );
  }
}